pvlib/iotools/psm3.py

```diff
@@ -0,0 +1,174 @@
+
+"""
+Get PSM3 TMY
+see https://developer.nrel.gov/docs/solar/nsrdb/psm3_data_download/
+"""
+
+import io
+import requests
+import pandas as pd
+# Python-2 compatible JSONDecodeError
+try:
+    from json import JSONDecodeError
+except ImportError:
+    JSONDecodeError = ValueError
+
+URL = "http://developer.nrel.gov/api/solar/nsrdb_psm3_download.csv"
+
+# 'relative_humidity', 'total_precipitable_water' are not available
+ATTRIBUTES = [
+    'air_temperature', 'dew_point', 'dhi', 'dni', 'ghi', 'surface_albedo',
+    'surface_pressure', 'wind_direction', 'wind_speed']
+PVLIB_PYTHON = 'pvlib python'
+
+
+def get_psm3(latitude, longitude, api_key, email, names='tmy', interval=60,
+             full_name=PVLIB_PYTHON, affiliation=PVLIB_PYTHON):
+    """
+    Get PSM3 data
+
+    Parameters
+    ----------
+    latitude : float or int
+        in decimal degrees, between -90 and 90, north is positive
+    longitude : float or int
+        in decimal degrees, between -180 and 180, east is positive
+    api_key : str
+        required NREL Developer Network API key
+    email : str
+        required, system uses this to automatically communicate messages back
+        to the user only if necessary
+    names : str
+        PSM3 API parameter specifing year or TMY variant to download, see notes
+        below for options, default is ``'tmy'``
```

Comment:
the "default is" pattern here and below is inconsistent with our current practice of putting "default X" next to the type spec. I don't care too much about consistency in `iotools`, so I'm ok with letting it go. Up to you or others.

Reply:
I'm hoping that's up for discussion, I don't want to break with convention, but I wish the default was not in the type spec, seems out of place there to me, but I could still be convinced otherwise - tell me to do it and I will :)

Reply:
I prefer (but not strongly) the `names : str, default ``'tmy'`` pattern, for the way it displays on readthedocs.

Reply:
Yes, up for discussion, no, not in this PR! Please make a new issue and collect some examples of projects that follow different standards.